app/liquid_tags/unified_embed.rb

```diff
@@ -1,46 +1,10 @@
-# The purpose of this singleton class is to provide a registry for the
-# numerous "link/embedded" type liquid tags.
-#
-# Each of those liquid tags must self-register their lookup regular
-# expression.
-class UnifiedEmbed
-  include Singleton
-
-  def self.current
-    instance
-  end
-
-  # @api public
-  #
-  # @param klass [Class] the LiquidTag class that we use when we match
-  #        the given :regexp
-  # @param regexp [Regexp] the regular expression that when matched
-  #        means we use the associated :klass
-  def self.register(klass, regexp:)
-    instance.register(klass, regexp: regexp)
-  end
-
-  # @api public
+# A namespacing module to help organize the concepts of embedding.
+module UnifiedEmbed
+  # A convenience method for registering tags as part of the
+  # UnifiedEmbed ecosystem.
   #
-  # @param link [String] the string that includes the URI for the
-  #        embed and possibly additional attributes, depending on how
-  #        the registered liquid tag parses this string.
-  # @return [Class] a descendant class of LiquidTagBase
-  def self.find_liquid_tag_for(link:)
-    instance.find_liquid_tag_for(link: link)
-  end
-
-  def initialize
-    @registry = []
-  end
-
-  def register(klass, regexp:)
-    @registry << [regexp, klass]
-  end
-
-  def find_liquid_tag_for(link:)
-    link = ActionController::Base.helpers.strip_tags(link).strip if link.include?("href=")
-    _regexp, klass = @registry.detect { |regexp, _tag_class| regexp.match?(link) }
-    klass
+  # @see UnifiedEmbed::Registry
+  def self.register(...)
+    UnifiedEmbed::Registry.register(...)
```

Comment:
since there's only a single static method there's no need for a singleton class here anymore, makes sense.

Reply:
Given my introduction of this indirection, we could remove the singleton structure from the Registry and instead favor a class instance variable for this module.

```ruby
def self.register(...)
  @registry ||= UnifiedEmbed::Registry.new
  @registery.register(...)
end
```

But what we have does appear to work.